portal-frontend/src/app/features/edit-application/nfu/nfu-proposal/nfu-proposal.component.ts

```diff
@@ -0,0 +1,152 @@
+import { Component, Input, OnDestroy, OnInit } from '@angular/core';
+import { FormControl, FormGroup } from '@angular/forms';
+import { MatButtonToggleChange } from '@angular/material/button-toggle';
+import { Router } from '@angular/router';
+import { BehaviorSubject, Subject, takeUntil } from 'rxjs';
+import { ApplicationDto, ApplicationUpdateDto } from '../../../../services/application/application.dto';
+import { ApplicationService } from '../../../../services/application/application.service';
+import { parseStringToBoolean } from '../../../../shared/utils/string-helper';
+
+@Component({
+  selector: 'app-nfu-proposal',
+  templateUrl: './nfu-proposal.component.html',
+  styleUrls: ['./nfu-proposal.component.scss'],
+})
+export class NfuProposalComponent implements OnInit, OnDestroy {
+  $destroy = new Subject<void>();
+  @Input() $application!: BehaviorSubject<ApplicationDto | undefined>;
+
+  hectares = new FormControl<number | null>(null);
+  purpose = new FormControl<string | null>(null);
+  outsideLands = new FormControl<string | null>(null);
+  agricultureSupport = new FormControl<string | null>(null);
+  willImportFill = new FormControl<string | null>(null);
+  totalFillPlacement = new FormControl<number | null>(null);
+  maxFillDepth = new FormControl<number | null>(null);
+  fillVolume = new FormControl<number | null>(null);
+  projectDurationYears = new FormControl<number | null>(null);
+  projectDurationMonths = new FormControl<number | null>(null);
+  fillTypeDescription = new FormControl<string | null>(null);
+  fillOriginDescription = new FormControl<string | null>(null);
+
+  form = new FormGroup({
+    hectares: this.hectares,
+    purpose: this.purpose,
+    outsideLands: this.outsideLands,
+    agricultureSupport: this.agricultureSupport,
+    willImportFill: this.willImportFill,
+    totalFillPlacement: this.totalFillPlacement,
+    maxFillDepth: this.maxFillDepth,
+    fillVolume: this.fillVolume,
+    projectDurationYears: this.projectDurationYears,
+    projectDurationMonths: this.projectDurationMonths,
+    fillTypeDescription: this.fillTypeDescription,
+    fillOriginDescription: this.fillOriginDescription,
+  });
+  private fileId: string | undefined;
+
+  constructor(private router: Router, private applicationService: ApplicationService) {}
+
+  ngOnInit(): void {
+    this.$application.pipe(takeUntil(this.$destroy)).subscribe((application) => {
+      if (application) {
+        this.fileId = application.fileNumber;
+
+        this.hectares.setValue(application.nfuHectares);
```

Comment:
just curious, why are you setting values one by one instead of patch?

Reply:
I forgot patch existed, updated